internal/service/projectipaddresses/data_source_test.go

```diff
@@ -0,0 +1,38 @@
+package projectipaddresses_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+)
+
+func TestAccProjectIPAddressesDS_basic(t *testing.T) {
+	var (
+		projectID      = acc.ProjectIDExecution(t)
+		dataSourceName = "data.mongodbatlas_project_ip_addresses.test"
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: ProjectIPAddressesConfig(projectID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet(dataSourceName, "project_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "services.clusters.#", "0"),
```

Comment:
could we have a test where a cluster is created in the project and it has some inbound and outbound ip addresses?

Reply:
have my doubts if we want to include a cluster creation/deletion for this, considering we do have a unit test that verifies correct handling of a complete response with ip addresses. No strong opinion, looping in @lantoli @maastha @EspenAlbert @marcosuma in case there is additional inputs.